src/gsy_e/models/strategy/profile.py

```diff
@@ -54,3 +91,12 @@ def read_or_rotate_profiles(self, reconfigure=False):
             profile_type=self.profile_type,
             profile=profile,
             profile_uuid=self.input_profile_uuid)
+
+
+def profile_factory(input_profile=None, input_profile_uuid=None,
+                    input_energy_rate=None,
+                    profile_type: InputProfileTypes = None) -> EnergyProfileBase:
+    """Return correct profile handling class for input parameters."""
+    return (EmptyProfile(profile_type)
+            if input_profile == input_profile_uuid == input_energy_rate is None
```

Comment:
I think that this is missing the CLI case, where the input profile is non-None, but the profile_uuid is None. Might be mistaken though, so if you could check whether gsy-e CLI works as expected with input profiles it would be great. Thanks!

Reply:
I will test, however, if the `input_profile` is not None, then it is good to select the `EnergyProfile` class, it deals with reading the in-setup-profiles too. 
Or did I get your concern wrong?